### ISSUE TITLE
Links that open in a new tab/window need to inform users a new tab/window will open

### ISSUE THREAD

**Issue by afercia** [WordPress/gutenberg#1105]
There are some cases where links use `target="_blank"`. For example:

<img width="292" alt="screen shot 2017-06-10 at 14 10 00" src="https://user-images.githubusercontent.com/1682452/27002643-f3ad1ee6-4de6-11e7-9cd2-a216b6fca466.png">

current link html:
`<a href="https://codex.wordpress.org/Excerpt" target="_blank">Learn more about manual excerpts</a>`

In all these cases, the link should inform users a new tab/window will open;
- for accessibility: some visually hidden text in the link text would be ok; core recently standardized to always use `(opens in a new window)`, see https://core.trac.wordpress.org/changeset/40643
- visually: I'll suggest to consider to use some visual indicator too, it's also a usability thing but this is definitely up to designers :)

See also: https://core.trac.wordpress.org/query?keywords=~target-blank

**Comment by aduth** [comment]
Could render an adjacent `<Dashicon icon="external" />` to visually indicate external: 

https://developer.wordpress.org/resource/dashicons/#external

Maybe even logic built-in to the link component to detect whether it should be shown by default based on `target` or "off-site" URL.

**Comment by afercia** [comment]
> Could render an adjacent <Dashicon icon="external" /> to visually indicate external:

Yes, something like that could help, however "external" doesn't exactly mean "new tab/window": it just means external site (as in opening an external site in the same window).

**Comment by mehigh** [comment]
I'm interested in working on this feature and improving the user experience with the addition of the external icon having an a11y label for the screen readers.

**Comment by mehigh** [comment]
Thank you @afercia and @aduth for the comments. They helped get this issue home quicker. PR created.

**Comment by swissspidy** [comment]
Most links in WordPress don't have `target="_blank"` though, including many links to the Codex.

**Comment by afercia** [comment]
Reopening because things have changed a bit and this issue still needs to be fully addressed.

On current master, there are a few occurrences of links that use a `target="_blank"` attribute. Worth reminding that _any_ link that opens a new tab / window needs to inform users about what is going to happen, to prevent an unexpected change of context.

Some of these links are:
- the Preview link 
- the link in the block toolbar when editing a link 
- the post permalink link 

Edge cases:
the links in the Categories and Latest Posts blocks use `target="_blank"` in the editor, so they need a proper informative message and the `rel` attribute with proper values, but they open in the current tab / window when in the front-end.

There is now an `ExternalLink` component, used for example in the Excerpt component for the link to the Codex page, but it uses also an icon so I'm not sure it can be used for the cases listed above.

<img width="293" alt="screen shot 2018-03-01 at 11 32 46" src="https://user-images.githubusercontent.com/1682452/36840942-05561cf8-1d47-11e8-95b5-3e18f755e338.png">

**Comment by karmatosed** [comment]
From a design perspective I don't mind adding this symbol:

![2018-07-05 at 22 17](https://user-images.githubusercontent.com/253067/42348396-351cf246-80a1-11e8-977c-87ff43f9f9c5.png)

If I am correct that's what is being asked for from design side?

**Comment by talldan** [comment]
I'll take a look at fixing this next.
